demos/rag_agentic/src/agent_with_mcp_ocp_slack.py

```diff
@@ -0,0 +1,110 @@
+# Code bellow written following examples here: https://llama-stack.readthedocs.io/en/latest/building_applications
+from llama_stack_client import Agent
+from llama_stack_client.lib.agents.event_logger import EventLogger
+from llama_stack_client import LlamaStackClient
+from termcolor import cprint
+import argparse
+import logging
+import os
+from dotenv import load_dotenv
+
+load_dotenv()
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+stream_handler = logging.StreamHandler()
+stream_handler.setLevel(logging.INFO)
+formatter = logging.Formatter('%(message)s')
+stream_handler.setFormatter(formatter)
+logger.addHandler(stream_handler)
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument("-r", "--remote", help="Uses the remote_url", action="store_true")
+parser.add_argument("-s", "--session-info-on-exit", help="Prints agent session info on exit", action="store_true")
+parser.add_argument("-a", "--auto", help="Automatically runs examples, and does not start a chat session", action="store_true")
+args = parser.parse_args()
+
+llama_model="meta-llama/Llama-3.2-3B-Instruct"
+granite_model="ibm-granite/granite-3.2-8b-instruct"
+# Connect to a llama stack server
+if args.remote:
+    base_url = os.getenv("REMOTE_BASE_URL")
+    mcp_url = os.getenv("REMOTE_MCP_URL")
+else:
+    base_url="http://localhost:8321"
+    mcp_url="http://host.containers.internal:8000/sse"
+
+client = LlamaStackClient(base_url=base_url)
+logger.info(f"Connected to Llama Stack server @ {base_url} \n")
+
+# Get tool info and register tools
+registered_tools = client.tools.list()
+registered_tools_identifiers = [t.identifier for t in registered_tools]
+registered_toolgroups = [t.toolgroup_id for t in registered_tools]
+
+if "mcp::slack" not in registered_toolgroups:
+    # Register MCP tools
+    client.toolgroups.register(
+        toolgroup_id="mcp::slack",
+        provider_id="model-context-protocol",
+        mcp_endpoint={"uri":mcp_url},
+        )
+mcp_tools = [t.identifier for t in client.tools.list(toolgroup_id="mcp::slack")]
+
+logger.info(f"""Your Server has access the the following toolgroups:
+{set(registered_toolgroups)}
+""")
+
+# system prompts for different models
+llama_prompt = """You are a helpful assistant. You have access to a number of tools.
+    Whenever a tool is called, be sure return the Response in a friendly and helpful tone."""
+granite_prompt = """You are a helpful AI assistant with access to the tools listed next. When a tool is required to answer the user's query, respond with `<tool_call>` followed by a JSON object of the tool used. For example: `<tool_call> {"name":"function_name","arguments":{"arg1":"value"}} </tool_call>`:The user will respond with the output of the tool execution response so you can continue with the rest of the initial user prompt (continue).
+If a tool does not exist in the provided list of tools, notify the user that you do not have the ability to fulfill the request."""
```

Comment:
Since `llama_model` and `llama_prompt` are currently hardcoded below, should we add parser argument to let users quickly switch between the two models and prompts?

Reply:
good idea, I can do that 👍